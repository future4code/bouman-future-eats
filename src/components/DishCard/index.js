import React from 'react';
import styled from 'styled-components'
import { updateSelectedDishes, removeDish } from '../../actions/orders'
import { connect } from 'react-redux';
import SelectQuantity, {TextButton} from '../SelectQuantity'


const MainContainer = styled.div`
  width: 328px;
  height: 112px;
  border-radius: 8px;
  border: solid 1px #b8b8b8;
  display:flex;
  position: relative;
  margin-bottom: 8px;
  `

const TextContainer = styled.div`
  margin-left: 16px;
  margin-top: 18px;
`

const DishMainName = styled.p`
  width: 167px;
  height: 18px;
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: -0.39px;
  color: #5cb646;
  margin: 0px;
`

const DishDescription = styled.p`
  width: 200px;
  height: 30px;
  font-family: Roboto;
  font-size: 12px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: -0.29px;
  color: #b8b8b8;
  margin: 0px;
  margin-top: 8px;
`
const Price = styled.p`
  width: 118px;
  height: 19px;
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: -0.39px;
  color: #000000;
  margin: 0px;
  margin-top: 4px;
`

const ButtonRemove = styled.button`
  width: 90px;
  height: 31px;
  border-top-left-radius: 8px;
  border-bottom-right-radius: 8px;
  border: solid 1px #e02020;
  background: white;
  position: absolute;
  right: 0px;
  bottom: 0px;
  display: flex;
  justify-content: center;
  p {
    color: #e02020
  }
`

const Quantity = styled.button`
  width: 40px;
  height: 31px;
  border-top-right-radius: 8px;
  border-bottom-left-radius: 8px;
  border: solid 1px #5cb646;
  background: white;
  position: absolute;
  right: 0px;
  top: 0px;
  display: flex;
  justify-content: center;
  p {
    color: #5cb646
  }
`


function DishCard(props) {

  const quantity = () => props.selectedDishes.filter(
    dish => dish.id === props.id
  ).map(
    dish => dish.quantity
  )

  return (
    <MainContainer>
      <img width='100' src={props.photoUrl} alt="PratoDoRestaurante" />
      <TextContainer>
        <DishMainName>{props.name}</DishMainName>
        <DishDescription>{props.description}</DishDescription>

        <Price>{`R$ ${props.price}`}</Price>
        {quantity().length !== 0 && (<Quantity><TextButton>{quantity()}</TextButton></Quantity>)}

        {quantity().length !== 0 ? 
        <ButtonRemove onClick={() => props.removeDish(props.id)}><p>Remover</p></ButtonRemove> : 
        <SelectQuantity updateSelectedDishes={props.updateSelectedDishes} id={props.id}/>}

      </TextContainer>
    </MainContainer>
  );
}

export default connect(
  state => ({
    selectedDishes: state.orders.selectedDishes
  }),
  dispatch => ({
    updateSelectedDishes: (id, quantity) => dispatch(updateSelectedDishes(id, quantity)),
    removeDish: (id) => dispatch(removeDish(id))
  })
)(DishCard)




