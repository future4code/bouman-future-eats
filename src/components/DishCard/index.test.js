import React from 'react'
import renderer from 'react-test-renderer'
import DishCard from './index'

test("Dish card snapshot", ( ) => {
    const tree= renderer.create(<DishCard text={''}/>).toJSON()

    expect(tree).toMatchSnapshot()
})