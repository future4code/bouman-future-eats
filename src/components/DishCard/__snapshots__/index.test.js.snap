// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dish card snapshot 1`] = `
<div
  className="sc-bdVaJa jYPUYi"
>
  <img
    alt="PratoDoRestaurante"
    src="dishImg.png"
  />
  <div
    className="sc-bwzfXH gkimBC"
  >
    <p
      className="sc-htpNat codcUx"
    >
      Stencil
    </p>
    <p
      className="sc-bxivhb kFZPIN"
    >
      PÃ£o, carne, queijo, cebola roxa, tomate, alface e molho
    </p>
    <p
      className="sc-ifAKCX kVjkDh"
    >
      R$ 67,70
    </p>
    <button
      className="sc-EHOje hhMRVz"
    >
      <p
        className="sc-gzVnrw zxOgP"
      >
        Adicionar
      </p>
    </button>
  </div>
</div>
`;
